spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraJoinRDD.scala

```diff
@@ -0,0 +1,152 @@
+
+  def on(joinColumns: ColumnSelector): CassandraJoinRDD[O, N] = {
+    new CassandraJoinRDD[O, N](prev, keyspaceName, tableName, connector, columnNames, joinColumns, where, readConf).asInstanceOf[this.type]
+  }
+
+  //We need to make sure we get selectedColumnNames before serialization so that our RowReader is
+  //built
+  private val singleKeyCqlQuery: (String) = {
+    require(tableDef.partitionKey.map(_.columnName).exists(
+      partitionKey => where.predicates.exists(_.contains(partitionKey))) == false,
+      "Partition keys are not allowed in the .where() clause of a Cassandra Join")
+    logDebug("Generating Single Key Query Prepared Statement String")
+    logDebug(s"SelectedColumns : $selectedColumnNames -- JoinColumnNames : $joinColumnNames")
+    val columns = selectedColumnNames.map(_.cql).mkString(", ")
+    val joinWhere = joinColumnNames.map(_.columnName).map(name => s"${quote(name)} = :$name")
+    val filter = (where.predicates ++ joinWhere).mkString(" AND ")
+    val quotedKeyspaceName = quote(keyspaceName)
+    val quotedTableName = quote(tableName)
+    val query = s"SELECT $columns FROM $quotedKeyspaceName.$quotedTableName WHERE $filter"
+    logDebug(query)
+    query
+  }
+
+  /**
+   * When computing a CassandraPartitionKeyRDD the data is selected via single CQL statements
+   * from the specified C* Keyspace and Table. This will be preformed on whatever data is
+   * avaliable in the previous RDD in the chain.
+   * @param split
+   * @param context
+   * @return
+   */
+  override def compute(split: Partition, context: TaskContext): Iterator[(O, N)] = {
+    logDebug(s"Query::: $singleKeyCqlQuery")
+    val session = connector.openSession()
+    implicit val pv = protocolVersion(session)
+    val stmt = session.prepare(singleKeyCqlQuery).setConsistencyLevel(consistencyLevel)
+    val bsb = new BoundStatementBuilder[O](rowWriter, stmt, pv)
+    val metricsUpdater = InputMetricsUpdater(context, 20)
+    val rowIterator = fetchIterator(session, bsb, prev.iterator(split, context))
+    val countingIterator = new CountingIterator(rowIterator)
+    context.addTaskCompletionListener { (context) =>
+      val duration = metricsUpdater.finish() / 1000000000d
+      logDebug(f"Fetched ${countingIterator.count} rows from $keyspaceName.$tableName for partition ${split.index} in $duration%.3f s.")
+      session.close()
+    }
+    countingIterator
+  }
+
+  private def fetchIterator(session: Session, bsb: BoundStatementBuilder[O], lastIt: Iterator[O]): Iterator[(O, N)] = {
+    val columnNamesArray = selectedColumnNames.map(_.selectedAs).toArray
+    implicit val pv = protocolVersion(session)
+    lastIt.map(leftSide => (leftSide, bsb.bind(leftSide))).flatMap { case (leftSide, boundStmt) =>
+      val rs = session.execute(boundStmt)
+      val iterator = new PrefetchingResultSetIterator(rs, fetchSize)
+      val result = iterator.map(rightSide => (leftSide, rowTransformer.read(rightSide, columnNamesArray)))
```

Comment:
Don't call it rowTransformer. This is probably a leftover from some refactoring and I we must have not noticed this name wasn't properly updated. Call it rowReader (as the class).